### PR TITLE
Flask + API

### PR DESCRIPTION
**Pull request by kipod**
* integration flask and fastapi
* build tags
* mypy

### CODE REVIEW

**Review by StoliarskyiKostiantyn — APPROVED** [review]

File: .github/workflows/build_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,135 @@
+name: Build and Deploy
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
```
Comment (DenisBurimov) [inline .github/workflows/build_deploy.yaml:6]: Shouldn't we specify branches?

File: .github/workflows/build_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+  deploy:
+    name: Deploy to staging
+    runs-on: Linux
+    needs:
+      - on-develop-branch-check
+      - build
+    # if: ${{ needs.on-develop-branch-check.outputs.on_develop == 'true' }}
+    if: true
```
Comment (DenisBurimov) [inline .github/workflows/build_deploy.yaml]: Do we deploy in any case?

File: .vscode/launch.json [hunk 3]
```diff
@@ -4,6 +4,25 @@
   // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
   "version": "0.2.0",
   "configurations": [
+    {
+      "name": "API",
+      "type": "python",
+      "request": "launch",
+      "module": "uvicorn",
+      "args": ["api:app", "--reload", "--port", "5002"],
+      "jinja": true,
```
Comment (DenisBurimov) [inline .vscode/launch.json]: Do we need jinja for an API if we have FLask?

File: .vscode/settings.json [hunk 4]
```diff
@@ -21,10 +23,10 @@
     "wsgi",
     "wtforms"
   ],
-  "python.testing.unittestArgs": ["-v", "-s", "./tests", "-p", "*test.py"],
+  // "python.testing.unittestArgs": ["-v", "-s", "./tests", "-p", "*test.py"],
```
Comment (DenisBurimov) [inline .vscode/settings.json]: Is this code redundant now?

File: .github/workflows/build_deploy.yaml (hunk 2 above)
Comment (kipod) [reply .github/workflows/build_deploy.yaml]: Yea, thanks for this note. Really, lets build only on develop branch.
FIXED

File: api/dependency/user.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        )
+    return user
+
+
+def get_user(request: Request, db: Session = Depends(get_db)) -> m.User | None:
+    """Raises an exception if the current user is authenticated"""
+    auth_header = request.headers.get("Authorization")
+    if auth_header:
+        # Assuming the header value is in the format "Bearer <token>"
+        token: s.TokenData = verify_access_token(
```
Comment (DenisBurimov) [inline api/dependency/user.py:48]: I suppose it would be better to check if Bearer is in the token

File: .vscode/launch.json (hunk 3 above)
Comment (kipod) [reply .vscode/launch.json]: you r right - no need

File: .vscode/settings.json (hunk 4 above)
Comment (kipod) [reply .vscode/settings.json]: correct

File: api/routes/auth.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+router = APIRouter(prefix="/auth", tags=["Auth"])
+
+
+@router.post("/login", status_code=status.HTTP_200_OK, response_model=s.Token)
+def login(
+    form_data: Annotated[OAuth2PasswordRequestForm, Depends()], db=Depends(get_db)
+):
+    """Logs in a user"""
+    user = m.User.authenticate(form_data.username, form_data.password, session=db)
+    if not user:
```
Comment (DenisBurimov) [inline api/routes/auth.py:21]: I think maybe we need a log in case of successful retrieving too

File: test_api/test.env [hunk 7]
```diff
@@ -0,0 +1,2 @@
+IS_API=true
+APP_ENV=testing
```
Comment (DenisBurimov) [inline test_api/test.env]: Enter

File: test_api/test_users.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+@pytest.mark.skipif(not CFG.IS_API, reason="API is not enabled")
+def test_get_me(
+    client: TestClient,
+    headers: dict[str, str],
+):
+    response = client.get("/api/users/me", headers=headers)
+    assert response.status_code == 200
+    user = s.User.model_validate(response.json())
+    assert user.username == "user1"
```
Comment (DenisBurimov) [inline test_api/test_users.py]: Maybe it would be better not to hardcode checking value

**Review by DenisBurimov — APPROVED** [review]

File: api/dependency/user.py (hunk 5 above)
Comment (kipod) [reply api/dependency/user.py:48]: added assertion

File: api/routes/auth.py (hunk 6 above)
Comment (kipod) [reply api/routes/auth.py:21]: +1

File: test_api/test_users.py (hunk 8 above)
Comment (kipod) [reply test_api/test_users.py]: OK :-)

**Pull request merged**